From b01f15afb00db88a459d0f9e02088db0f687bd72 Mon Sep 17 00:00:00 2001
From: Gustavo Duarte <gduarte@redhat.com>
Date: Fri, 1 Aug 2014 00:50:22 -0400
Subject: [powerpc] perf: Reject EBB events which specify a sample_type

Message-id: <1406854226-17977-2-git-send-email-gduarte@redhat.com>
Patchwork-id: 86927
O-Subject: [RHEL7.1 PATCH BZ 1121408 1/5] powerpc/perf: Reject EBB events which specify a sample_type
Bugzilla: 1121408
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1121408
Upstream Status: 58b5fb00498ddaaa3f6582751e9eb731189ee4c8

commit 58b5fb00498ddaaa3f6582751e9eb731189ee4c8
Author: Michael Ellerman <mpe@ellerman.id.au>
Date:   Fri Mar 14 16:00:30 2014 +1100

    powerpc/perf: Reject EBB events which specify a sample_type

    Although we already block EBB events which request sampling using
    sample_period, technically it's possible for an event to set sample_type
    but not sample_period.

    Nothing terrible will happen if an EBB event does specify sample_type,
    but it signals a major confusion on the part of userspace, and so we do
    them the favor of rejecting it.

    Signed-off-by: Michael Ellerman <mpe@ellerman.id.au>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/perf/core-book3s.c b/arch/powerpc/perf/core-book3s.c
index 06433ba..39af8d8 100644
--- a/arch/powerpc/perf/core-book3s.c
+++ b/arch/powerpc/perf/core-book3s.c
@@ -504,8 +504,11 @@ static int ebb_event_check(struct perf_event *event)
   if (!leader->attr.pinned || !leader->attr.exclusive)
    return -EINVAL;
 
-  if (event->attr.inherit || event->attr.sample_period ||
-      event->attr.enable_on_exec || event->attr.freq)
+  if (event->attr.freq ||
+      event->attr.inherit ||
+      event->attr.sample_type ||
+      event->attr.sample_period ||
+      event->attr.enable_on_exec)
    return -EINVAL;
  }
 
-- 
1.7.1